### PR TITLE
Update Score-P to 2.0

### PR DESCRIPTION
**Pull request by suraia**
This PR updates Score-P to version 2.0. It also updates its dependencies to the appropriate versions.

### CODE REVIEW

**Comment by becker33** [comment]
Please fix the flake8 errors

**Comment by becker33** [comment]
Does Score-P depend specifically on the newest version of these packages, or can it link with any version of its dependencies?

File: var/spack/repos/builtin/packages/scorep/package.py [hunk 1]
```diff
@@ -56,13 +64,14 @@ class Scorep(Package):
     depends_on("papi")
 
     def install(self, spec, prefix):
-        configure = Executable( join_path(self.stage.source_path, 'configure') )
+        configure = Executable(join_path(self.stage.source_path, 'configure'))
         with working_dir('spack-build', create=True):
             configure_args = ["--prefix=%s" % prefix,
```
Comment (eschnett) [inline var/spack/repos/builtin/packages/scorep/package.py]: For future reference: I like to break lines right after the opening `[`, as this tends to lead reduce the indentation of the whole list, and one doesn't need to split individual list items over multiple lines.

**Comment by suraia** [comment]
According to the website (http://www.vi-hps.org/projects/score-p/) it "is supposed to work with" the newest versions. I have not actually tried linking it with older versions.

I have also fixed the the list indentation to address @eschnett's comment.

**Pull request merged**